File: frameworks/ruby-rails-fixed/app/graphql/types/user_type.rb
module Types
  class UserType < Types::BaseObject
    field :id, ID, null: false
    field :username, String, null: false
    field :full_name, String, null: true
    field :bio, String, null: true

    field :posts, [Types::PostType], null: false
    field :comments, [Types::CommentType], null: false

    def posts
      Post.where(fk_author: object.pk_user)
    end

    def comments
      Comment.where(fk_author: object.pk_user)
    end
  end
end

File: frameworks/csharp-dotnet/FraiseQL.Benchmark/GraphQL/Types/UserType.cs
using FraiseQL.Benchmark.Models;
using HotChocolate.Types;

namespace FraiseQL.Benchmark.GraphQL.Types;

public class UserType : ObjectType<User>
{
    protected override void Configure(IObjectTypeDescriptor<User> descriptor)
    {
        descriptor.Field(u => u.Id);
        descriptor.Field(u => u.Username);
        descriptor.Field(u => u.FullName);
        descriptor.Field(u => u.Bio);
        descriptor.Field(u => u.Posts);
        descriptor.Field(u => u.Comments);
    }
}